scripts/tests/run_python_test.py

```diff
@@ -0,0 +1,160 @@
+        if path.name != name:
+            continue
+        return str(path)
+
+    return None
+
+
+def EnqueueLogOutput(fp, tag, q):
+    for line in iter(fp.readline, b''):
+        timestamp = time.time()
+        if len(line) > len('[1646290606.901990]') and line[0:1] == b'[':
+            try:
+                timestamp = float(line[1:18].decode())
+                line = line[19:]
+            except Exception as ex:
+                pass
+        q.put((tag, line, datetime.datetime.fromtimestamp(
+            timestamp).isoformat(sep=" ")))
+    fp.close()
+
+
+def RedirectQueueThread(fp, tag, queue) -> threading.Thread:
+    log_queue_thread = threading.Thread(target=EnqueueLogOutput, args=(
+        fp, tag, queue))
+    log_queue_thread.start()
+    return log_queue_thread
+
+
+def DumpLogOutput(q: queue.Queue, timeout: int = 0):
+    deadline = time.time() + timeout
+    while True:
+        line = q.get((time.time() - deadline) > 0,
+                     max(deadline - time.time(), 0))
+        sys.stdout.buffer.write(
+            (f"[{line[2]}]").encode() + line[0] + line[1])
+        sys.stdout.flush()
+
+
+@click.command()
+@click.option("--app", type=str, default=None, help='Local application to use, omit to use external apps.')
+@click.option("--factoryreset", is_flag=True, help='Remove /tmp/chip* before running the tests.')
+@click.option("--wait-before-test", type=int, default=3, help='Time for the device to start before running the tests.')
+@click.option("--app-params", type=str, default='', help='The extra parameters passed to the device.')
+@click.option("--script", type=click.Path(exists=True), default=FindBinaryPath("mobile-device-test.py"), help='Test script to use.')
+@click.argument("script-args", nargs=-1, type=str)
+def main(app: str, factoryreset: bool, wait_before_test: int, app_params: str, script: str, script_args: typing.List[str]):
+    if factoryreset:
+        retcode = subprocess.call("rm -rf /tmp/chip*", shell=True)
+        if retcode != 0:
+            raise Exception("Failed to remove /tmp/chip* for factory reset.")
+
+    log_queue = queue.Queue()
+    log_cooking_threads = []
+
+    app_process = None
+    if app:
+        app = FindBinaryPath(app)
+        if app is None:
+            raise FileNotFoundError(f"{app} not found")
+        app_args = [app] + shlex.split(app_params)
+        logging.info(f"Execute: {app_args}")
+        app_process = subprocess.Popen(
+            app_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=0)
+        log_cooking_threads.append(RedirectQueueThread(app_process.stdout,
+                                                       b"[\33[34mAPP \33[0m][\33[33mSTDOUT\33[0m]", log_queue))
+        log_cooking_threads.append(RedirectQueueThread(app_process.stderr,
+                                                       b"[\33[34mAPP \33[0m][\33[31mSTDERR\33[0m]", log_queue))
+
+    try:
+        DumpLogOutput(log_queue, wait_before_test)
```

Comment:
Doesn't this block till all the logs have been flushed BEFORE it continues and executes the next process execution?

Reply:
It will block for the specificed timeout, this is removed since the sleep is not necessary